orc/src/main/java/org/apache/iceberg/orc/OrcSchemaWithTypeVisitor.java

```diff
@@ -78,14 +79,73 @@ protected T visitUnion(Type type, TypeDescription union, OrcSchemaWithTypeVisito
     if (types.size() == 1) { // single type union
       options.add(visit(type, types.get(0), visitor));
     } else { // complex union
-      for (int i = 0; i < types.size(); i += 1) {
-        options.add(visit(type.asStructType().fields().get(i + 1).type(), types.get(i), visitor));
-      }
+      visitComplexUnion(type, union, visitor, options);
     }
 
     return visitor.union(type, union, options);
   }
 
+  /*
+  A complex union with multiple types of Orc schema is converted into a struct with multiple fields of Iceberg schema.
+  Also an extra tag field is added into the struct of Iceberg schema during the conversion.
+  Given an example of complex union in both Orc and Iceberg:
+  Orc schema: {"name":"unionCol","type":["int","string"]}
```

Comment:
Lets add a 1 liner clarifying that 'nulls' and 'defaults' are not a consideration for ORC schemas.

Reply:
Good point, done